### PR TITLE
[devops #1764044] unify dependency management for `azure-maven-plugins`

### PR DESCRIPTION
**Pull request by wangmingliang-ms**
Task 1769084: manage dependencies in root project's pom.

### CODE REVIEW

File: azure-tools-common/pom.xml [hunk 1]
```diff
@@ -85,87 +58,74 @@
...
             <groupId>com.fasterxml.jackson.core</groupId>
             <artifactId>jackson-databind</artifactId>
-            <version>${jackson.core.version}</version>
         </dependency>
         <dependency>
             <groupId>commons-io</groupId>
             <artifactId>commons-io</artifactId>
-            <version>${commons.io.version}</version>
+        </dependency>
+        <dependency>
```
Comment (Flanker32) [inline azure-tools-common/pom.xml:90]: Is there any code changes to add a new dependency here?

File: azure-webapp-maven-plugin/pom.xml [hunk 2]
```diff
@@ -317,30 +281,6 @@
                     </execution>
                 </executions>
             </plugin>
-            <plugin>
```
Comment (Flanker32) [inline azure-webapp-maven-plugin/pom.xml:320]: https://github.com/microsoft/azure-maven-plugins/pull/81, this is for better development experience in Eclipse, do we need to remove it?

File: azure-tools-common/pom.xml (hunk 1 above)
Comment (wangmingliang-ms) [reply azure-tools-common/pom.xml:90]: no, but it's explicitly used in code. no sure why it's not explicitly declared.
```
com/microsoft/azure/common/function/handlers/runtime/DockerFunctionRuntimeHandler.java:18
```
```
com/microsoft/azure/common/utils/TextUtilsTest.java:8
```

File: azure-webapp-maven-plugin/pom.xml (hunk 2 above)
Comment (wangmingliang-ms) [reply azure-webapp-maven-plugin/pom.xml:320]: I will restore it.

**Review by andxu — APPROVED** [review]

**Pull request merged**